rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
Because we need the client to have a `view_id` before we send it notifications about running plugins and available commands, etc. I'm very open to a better strategy if you have any ideas (although I think in general the pattern of doing plugin maintenance etc work in the idle call will become more useful)